From d5ac61420cce02c334511ed98757924ef53675e9 Mon Sep 17 00:00:00 2001
From: Dhana Babu <dhana36.m@gmail.com>
Date: Mon, 26 Jul 2021 13:20:47 -0400
Subject: [PATCH] Fix: Working outside of request context

---
 flask_restx/marshalling.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/flask_restx/marshalling.py b/flask_restx/marshalling.py
index 8e899f7f..a856af42 100644
--- a/flask_restx/marshalling.py
+++ b/flask_restx/marshalling.py
@@ -5,7 +5,7 @@
 from functools import wraps
 from six import iteritems
 
-from flask import request, current_app, has_app_context
+from flask import request, current_app, has_app_context, has_request_context
 
 from .mask import Mask, apply as apply_mask
 from .utils import unpack
@@ -247,7 +247,7 @@ def __call__(self, f):
         def wrapper(*args, **kwargs):
             resp = f(*args, **kwargs)
             mask = self.mask
-            if has_app_context():
+            if has_app_context() and has_request_context():
                 mask_header = current_app.config["RESTX_MASK_HEADER"]
                 mask = request.headers.get(mask_header) or mask
             if isinstance(resp, tuple):
