https://gitlab.com/desktop-frameworks/wayqt/-/issues/6

diff --git a/src/SessionLockImpl.cpp b/src/SessionLockImpl.cpp
index a5f2e84..74fd5c1 100644
--- a/src/SessionLockImpl.cpp
+++ b/src/SessionLockImpl.cpp
@@ -78,5 +78,9 @@ void SessionLockSurfaceImpl::configureCallback( void *data, struct ::ext_session
     #endif
 
     /** Commit the buffer */
+    #if QT_VERSION >= QT_VERSION_CHECK( 6, 10, 0 )
+    sls->window()->updateExposure();
+    #else
     sls->handleExpose( QRect( QPoint(), QSize((int)width, (int)height ) ) );
+    #endif
 }

